run.sh

```diff
@@ -1,20 +1,63 @@
 #!/bin/bash
 
-access_id=${SUMO_ACCESS_ID:=$1}
-access_key=${SUMO_ACCESS_KEY:=$2}
-receiver_url=${SUMO_RECEIVER_URL:=https://collectors.sumologic.com}
-collector_name=${SUMO_COLLECTOR_NAME:=collector_container}
-sources_json=${SUMO_SOURCES_JSON:=/etc/sumo-sources.json}
+SUMO_ACCESS_ID=${SUMO_ACCESS_ID:=$1}
+SUMO_ACCESS_KEY=${SUMO_ACCESS_KEY:=$2}
+SUMO_RECEIVER_URL=${SUMO_RECEIVER_URL:=https://collectors.sumologic.com}
+SUMO_COLLECTOR_NAME=${SUMO_COLLECTOR_NAME_PREFIX:=''}${SUMO_COLLECTOR_NAME:=`cat /etc/hostname`}
+SUMO_SOURCES_JSON=${SUMO_SOURCES_JSON:=/etc/sumo-sources.json}
+SUMO_SYNC_SOURCES=${SUMO_SYNC_SOURCES:=false}
 
-if [ -z "$access_id" ] || [ -z "$access_key" ]; then
+if [ -z "$SUMO_ACCESS_ID" ] || [ -z "$SUMO_ACCESS_KEY" ]; then
 	echo "FATAL: Please provide credentials, either via the SUMO_ACCESS_ID and SUMO_ACCESS_KEY environment variables,"
 	echo "       or as the first two command line arguments!"
 	exit 1
 fi
 
-if [ ! -e $sources_json ]; then
-	echo "FATAL: Unable to find $sources_json - please make sure you include it in your image!"
+if [ ! -e "${SUMO_SOURCES_JSON}" ]; then
+	echo "FATAL: Unable to find $SUMO_SOURCES_JSON - please make sure you include it in your image!"
 	exit 1
 fi
 
-/opt/SumoCollector/collector console -- -t -i $access_id -k $access_key -n $collector_name -s $sources_json -u $receiver_url
+if [ "${SUMO_SYNC_SOURCES}" == "true" ]; then
+    SUMO_SYNC_SOURCES=${SUMO_SOURCES_JSON}
+    unset SUMO_SOURCES_JSON
+else
+    unset SUMO_SYNC_SOURCES
+fi
+
+# Supported user.properties configuration parameters
+# More information https://help.sumologic.com/Send_Data/Installed_Collectors/stu_user.properties
+declare -A SUPPORTED_OPTIONS
+SUPPORTED_OPTIONS=(
+    ["SUMO_ACCESS_ID"]="accessid"
+    ["SUMO_ACCESS_KEY"]="accesskey"
+    ["SUMO_RECEIVER_URL"]="url"
+    ["SUMO_COLLECTOR_NAME"]="name"
+    ["SUMO_SOURCES_JSON"]="sources"
+    ["SUMO_SYNC_SOURCES"]="syncSources"
+    ["PROXY_HOST"]="proxyHost"
+    ["PROXY_PORT"]="proxyPort"
+    ["PROXY_USER"]="proxyUser"
+    ["PROXY_PASSWORD"]="proxyPassword"
+    ["PROXY_NTLM_DOMAIN" ]="proxyNtlmDomain"
+    ["CLOBBER"]="clobber"
```

Comment:
Ah yeah, you're right it was using the (undocumented) command line argument `-t` to make it ephemeral.   

In that case, I think we need to preserve this behavior, especially since the docker container is ephemeral.  Making the collector ephemeral ensures that it will be removed from our backend after a period of inactivity (and prevent "leaking" collectors).  So maybe we should set `ephemeral=true` by default in user.properties.